aydin/nn/models/torch/test/test_torch_unet.py

```diff
@@ -45,27 +32,32 @@ def test_supervised_2D_n2t():
     input_image = torch.tensor(input_image)
     lizard_image = torch.tensor(lizard_image)
 
-    # dataset = TorchDataset(input_image, lizard_image, 64, self_supervised=False)
+    model = UNetModel(nb_unet_levels=2, supervised=True, spacetime_ndim=2)
 
-    # data_loader = DataLoader(
-    #     dataset, batch_size=1, shuffle=True, num_workers=0, pin_memory=True
-    # )
+    n2t_unet_train_loop(input_image, lizard_image, model)
 
-    model = UNetModel(
-        nb_unet_levels=2, supervised=True, spacetime_ndim=2, residual=True
-    )
+    denoised = model(input_image)
 
-    n2t_unet_train_loop(input_image, lizard_image, model)
+    if visualize:
+        import napari
+
+        viewer = napari.Viewer()
+        viewer.add_image(to_numpy(lizard_image), name="groundtruth")
+        viewer.add_image(to_numpy(input_image), name="noisy")
+        viewer.add_image(to_numpy(denoised), name="denoised")
+
+        napari.run()
 
     # assert result.shape == input_image.shape
     # assert result.dtype == input_image.dtype
 
 
-def test_masking_2D():
-    input_array = torch.zeros((1, 1, 64, 64))
+@pytest.mark.parametrize("nb_unet_levels", [2, 3, 5, 8])
```

Comment:
8 layers?!

Reply:
Might worth to test, don't you think so? :)